pahfit/base.py

```diff
@@ -0,0 +1,462 @@
+from astropy.modeling.functional_models import Gaussian1D
+
+from pahfit.component_models import (BlackBody1D, Drude1D,
+                                     S07_attenuation)
+
+from astropy.table import Table, vstack
+
+import numpy as np
+
+__all__ = ['PAHFITBase']
+
+
+def _ingest_limits(min_vals, max_vals):
+    """
+    Ingest the limits read from a file and generate the appropriate
+    internal format (list of tuples).  Needed to handle the case
+    where a limit is not desired as numpy arrays cannot have elements
+    of None, instead a value of nan is used.
+
+    Limits that are not set are designated as 'nan' in files and
+    these are changed to the python None to be compatible with
+    the astropy.modeling conventation.
+
+    Parameters
+    ----------
+    min_vals,
+    max_vals : numpy.array
+        min/max values of the limits for a parameter
+        nan designates no limit
+
+    Returns
+    -------
+    plimits : list of tuples
+        tuples give the min/max limits for a parameter
+    """
+    plimits = []
+    for cmin, cmax in zip(min_vals, max_vals):
+        if np.isnan(cmin):
+            cmin = None
+        if np.isnan(cmax):
+            cmax = None
+        plimits.append((cmin, cmax))
+
+    return plimits
+
+
+def _ingest_fixed(fixed_vals):
+    """
+    Ingest the fixed value read from a file and generate the appropriate
+    internal format (list of booleans).  Needed as booleans are
+    represented in files as True/False strings.
+
+    Parameters
+    ----------
+    min_vals : numpy.array (string)
+        fixed designations
+
+    Returns
+    -------
+    pfixed : numpy.array (boolean)
+        True/False designation for parameters
+    """
+    pfixed = []
+    for cfixed in fixed_vals:
+        if cfixed == 'True':
+            cfixed = True
+        if cfixed == 'False':
+            cfixed = False
+        pfixed.append(cfixed)
+
+    return pfixed
+
```

Comment:
Do we also need to have a variable for the name of the feature that we want to set limits or fixed values, or the input file should contain all quantities in a given order ?

Reply:
The input contains everything needed.  The two helper functions at the top base.py provide translation between the on disk format (limitations) and the in memory format needed.